src/Bicep.LangServer.IntegrationTests/InputOutputTests.cs

```diff
@@ -0,0 +1,256 @@
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.IO.Pipes;
+using System.Net;
+using System.Net.Sockets;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Bicep.Core.Extensions;
+using Bicep.Core.FileSystem;
+using Bicep.Core.Navigation;
+using Bicep.Core.Parsing;
+using Bicep.Core.Samples;
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using Bicep.Core.Syntax.Visitors;
+using Bicep.Core.Text;
+using Bicep.Core.TypeSystem.Az;
+using Bicep.Core.UnitTests;
+using Bicep.Core.UnitTests.Assertions;
+using Bicep.Core.UnitTests.Utils;
+using Bicep.Core.Workspaces;
+using Bicep.LangServer.IntegrationTests.Assertions;
+using Bicep.LangServer.IntegrationTests.Extensions;
+using Bicep.LangServer.IntegrationTests.Helpers;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using OmniSharp.Extensions.LanguageServer.Client;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client;
+using OmniSharp.Extensions.LanguageServer.Protocol.Document;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using OmniSharp.Extensions.LanguageServer.Protocol.Window;
+using SymbolKind = Bicep.Core.Semantics.SymbolKind;
+
+namespace Bicep.LangServer.IntegrationTests
+{
+    [TestClass]
+    public class InputOutputTests
+    {
+        [NotNull]
+        public TestContext? TestContext { get; set; }
+
+        private CancellationToken GetCancellationTokenWithTimeout(TimeSpan timeSpan)
+            => CancellationTokenSource.CreateLinkedTokenSource(
+                new CancellationTokenSource(timeSpan).Token,
+                TestContext.CancellationTokenSource.Token).Token;
+
+        private static Process StartServerProcessWithConsoleIO()
+        {
+            var exePath = typeof(LanguageServer.Program).Assembly.Location;
+
+            var process = new Process
+            {
+                StartInfo = new ProcessStartInfo
+                {
+                    FileName = "dotnet",
+                    Arguments = exePath,
+                    UseShellExecute = false,
+                    RedirectStandardError = true,
+                    RedirectStandardOutput = true,
+                    RedirectStandardInput = true,
+                },
+            };
+
+            process.Start();
+
+            return process;
+        }
+
+        private static Process StartServerProcessWithNamedPipeIo(string pipeName)
+        {
+            var exePath = typeof(LanguageServer.Program).Assembly.Location;
+
+            var process = new Process
+            {
+                StartInfo = new ProcessStartInfo
+                {
+                    FileName = "dotnet",
+                    Arguments = $"{exePath} --pipe {pipeName}",
+                    UseShellExecute = false,
+                    RedirectStandardError = true,
```

Comment:
>RedirectStandardError [](http://example.com/codeflow?start=20&length=21)

If you don't have something to consume the stderr and stdout, you can get a deadlock. (Easiest is just to add dummy event handlers with the async redirection.)